packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -1086,15 +1094,28 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     if (handlePath.length > 0 && !handlePath.startsWith("/")) {
                         handlePath = `/${handlePath}`;
                     }
-                    id = `${parentHandle}/.app${handlePath}`;
-
-                    // TODO: SPO will deprecate this soon
-                    if (summaryObject.handleType === api.SummaryType.Commit) {
-                        value = {
-                            content: id,
-                        };
+                    const pathKey = `.app${handlePath}`;
+                    const summaryTreeToExpand = this.treesPathToSummaryTreeCache.get(pathKey);
+                    if (summaryTreeToExpand !== undefined) {
+                        const result = await this.convertSummaryToSnapshotTree(
+                            parentHandle,
+                            summaryTreeToExpand,
+                            blobsShaToPathCacheLatest,
+                            treesPathToSummaryTreeCacheLatest,
+                            depth + 1,
+                            `${path}/${key}`);
+                        value = result.snapshotTree;
+                        reusedBlobs += result.reusedBlobs;
+                        blobs += result.blobs;
+                    } else {
+                        id = `${parentHandle}/.app${handlePath}`;
+                        // TODO: SPO will deprecate this soon
+                        if (summaryObject.handleType === api.SummaryType.Commit) {
```

Comment:
Is this dead code? Can it be removed?

Reply:
No it is not dead code because it may happen that the tree is not found in cache and then we have to use the handle.